fasort.sh

```diff
@@ -2,21 +2,21 @@
 # Author: Noxy (a dhole who likes paws)
 # This script will find and move any FA files in the working directory and move them to artist folders under $SORTDIR.
 
+set -euo pipefail
+
 # Directory where you want files to end up
 SORTDIR=~/Pictures/FA_Sorted
 
 # Find files in pwd that match what I think is a decent regex for FA image files
-for f in `ls | egrep '^\d+\.[0-9a-zA-Z-]+_.+\.\w+$'`
+find . -iregex '.*/[0-9]+\.[a-z0-9]+_.*' -print0 | while read -d $'\0' f
```

Comment:
Using find + read is better because: 

* Filenames can contain spaces and even newlines
* No subshell is created and kept running while it is looping through the results
* The `find` command is much more powerful than `ls`